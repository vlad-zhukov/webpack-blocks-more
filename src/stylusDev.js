/*
 * Stylus dev loader.
 */
export default function stylusDev(options = {}) {
    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'stylus-loader',
                    options: {
                        compress: false,
                        'disable-cache': true,
                        preferPathResolver: 'webpack',
                        ...options,
                    },
                },
            ],
        });
}
