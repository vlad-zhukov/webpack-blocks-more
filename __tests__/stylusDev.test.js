import {createConfig, match} from '@webpack-blocks/webpack';
import {css} from '@webpack-blocks/assets';
import stylusDev from '../src/stylusDev';

it('should work with match()', () => {
    const config = createConfig([match('*.styl', [stylusDev()])]);

    expect(config).toMatchSnapshot();
});

it('should take options', () => {
    const config = createConfig([
        match('*.styl', [
            stylusDev({
                compress: true,
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});

it('should work with css block', () => {
    const config = createConfig([match('*.styl', [css(), stylusDev()])]);

    expect(config).toMatchSnapshot();
});
