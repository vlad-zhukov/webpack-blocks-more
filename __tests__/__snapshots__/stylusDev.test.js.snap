// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": true,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with css block 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          "style-loader",
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": false,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.styl\\$/,
        "use": Array [
          Object {
            "loader": "stylus-loader",
            "options": Object {
              "compress": false,
              "disable-cache": true,
              "preferPathResolver": "webpack",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
